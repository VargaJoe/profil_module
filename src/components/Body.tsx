import * as React				            from 'react';
import { connect }                          from 'react-redux';
import Sidebar                              from './Sidebar';
import Header                               from './Header';
import {
    Route,
    Redirect,
    Switch,
    withRouter,
}                                           from 'react-router-dom';
import { LoginState }                       from '@sensenet/client-core';
import { Actions }                			from '@sensenet/redux';
import { PathHelper }                       from '@sensenet/client-utils';
import OtherUser                            from './OtherUser';
import Profil                               from './Profil';
import EditProfil                           from './EditProfil';

const DATA = require('../config.json');

class Body extends React.Component<any, any> {

    constructor(props: any) {
        super(props);
        this.state = {
            open : false,
            isDataFetched: false
        },

        this.openMenu = this.openMenu.bind(this);
    }

    openMenu() {
        let menuState = !this.state.open;
        this.setState({         
            open: menuState,
            articles: {}
          }); 
    } 

    componentDidMount  () {
        let path = PathHelper.joinPaths(DATA.home);
		// get the current user info
		let userGet = this.props.getHomeContent(path, {
            query: 'TypeIs:KnowledgeBaseArticle_v_2',
		});
		console.log(userGet);
        
        userGet.then( (result: any) => {
				this.setState({ 
                    isDataFetched : true,
					articles: result.value.entities.entities
				});
        });

        userGet.catch((err: any) => {
            console.log(err);
		});
	}

    render() {
        if ( !this.state.isDataFetched ) {
            return null;
		}
        const status = this.props.userLoginState !== LoginState.Authenticated;
        
        // let homePageItems = this.state.articles;
        // const homePage = Object.keys(homePageItems).map( (key: any) => 
        //     (
        //         <div key={key}>
        //             <h2>{homePageItems[key].DisplayName}</h2>
        //             <div>{homePageItems[key].Description}</div>
        //         </div>
        //     )
        // );

		return (
			<div className={this.state.open ? 'content_to_right open' : 'content_to_right'}>
                <Header />
                <Sidebar openMenu={this.props.openMenu}/>
                <div className="sn_overflow" onClick={this.props.openMenu} />
                <main className="sn_main">
                    <div className="sn_wrapp">
                        {/* <h1>Home Page</h1>
                        {homePage} */}
                        <Switch>
                            <Route 
                                exact={true}
                                path="/otherUser"  
                                render={(routerProps) => {
                                    return status ?
                                    <Redirect key="login" to="/login" />
                                    : <OtherUser {...routerProps} />;
                                }} 
                            />
                            <Route 
                                exact={true}
                                path="/user/:user"
                                render={(routerProps) => {
                                    return status ?
                                    <Redirect key="login" to="/login" />
                                    : <Profil {...routerProps} />;
                                }} 
                                userName={this.props.userName}
                            />
                            <Route 
                                exact={true}
                                path="/editUser"  
                                render={(routerProps) => {
                                    return status ?
                                    <Redirect key="login" to="/login" />
                                    : <EditProfil {...routerProps} />;
                                }} 
                            />
                        </Switch> 
                    </div>
                </main>
            </div>
		);
	}
}

const mapStateToProps = (state: any, match: any) => {
    return {
        userName :              state.sensenet.session.user.userName,
        userLoginState: 		state.sensenet.session.loginState, // state.user.user.FullName,
    };
};

export default withRouter(connect(
    mapStateToProps,
    (dispatch) => ({
        getHomeContent:    (path: string, options: any) => dispatch(Actions.requestContent( path, options )),
    })
)(Body as any));
 