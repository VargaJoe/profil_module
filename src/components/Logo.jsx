import React, { Component } 		from 'react';
import logo 						from '../images/logo.png';
import { Link } 					from 'react-router-dom'

class Logo extends Component {
	render () {
		return (
			<div className="sidebar__logo">
				<Link to="/">
					<img src={logo} alt="site logo" />
				</Link>
			</div>
		);
	}
};

export default Logo;
