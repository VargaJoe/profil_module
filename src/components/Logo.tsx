import React				 		from 'react';
import logo 						from '../images/logo.png';
import { Link } 					from 'react-router-dom';

export interface MenuTrigger { 
	openMenu: boolean;
}

class Logo extends React.Component<MenuTrigger, {}> {
	constructor(props: MenuTrigger) {
        super(props);
	}
	
	render () {
		return (
			<div className="sn_sidebar__logo">
				<Link to="/">
					<img src={logo} alt="site logo" />
				</Link>
				<div className="sn_menu" onClick={this.props.openMenu}>
					<span></span>
				</div>
			</div>
		);
	}
};

export default Logo;
