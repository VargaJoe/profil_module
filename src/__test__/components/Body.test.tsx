jest.unmock('../../index.tsx');
jest.unmock('redux-mock-store');
jest.unmock('../../components/Body');

import * as React 						        from 'react';
import Body, { BodyComponent } 					from '../../components/Body';
import Logo 							        from '../../components/Logo';
import { 
	configure, 
    shallow,
    mount, 
    ShallowWrapper, 
    ReactWrapper, 
    render } 							        from 'enzyme';
import * as Adapter 					        from 'enzyme-adapter-react-16';
import toJson 				      		        from 'enzyme-to-json';
import {
    withRouter,
    BrowserRouter as Router
}                                               from 'react-router-dom';
import * as createMockStore						from 'redux-mock-store';
import { Provider } 							from 'react-redux';

configure( {adapter: new Adapter()} );

const DATA = require('../../config.json');

describe('<Body /> mount rendering when data is added', () => {
    let store, 
        bodyMount: ReactWrapper<any, any>,
        bodyComponentShallow: ShallowWrapper<any, any>;
        
	const mockStore = (createMockStore as any)();
	beforeEach( () => {
        store = mockStore({
			sensenet: {
				session: {
					loginState: 'Unauthenticated',
					user: {
						userName: 'Visitor'
					}
				}
            },
            user: {
				user: {

				}
			} 
        });
        const props = {
			loginState:     'Unauthenticated',
			userName :      'Visitor'
        }; 
        bodyMount = mount( (
            <Router>
                <Provider store={store}>
                    <Body {...props} />
                </Provider>
            </Router>)); 
        bodyComponentShallow = shallow( (
            // <Router>
            //     <Provider store={store}>
                    <BodyComponent {...props} />
            //     </Provider>
            // </Router>
        )); 
	});   

    // test Snapshot 
	it('Match to snapshot', () => {
        expect(toJson(bodyMount)).toMatchSnapshot();
        expect(toJson(bodyComponentShallow)).toMatchSnapshot();
    });

    it('Match to snapshot with status flase', () => {
        bodyMount.setState({
            status: true
        });
        expect(toJson(bodyMount)).toMatchSnapshot();
    });

    // test Snapshot 
	it('Match to snapshot with status true', () => {
        bodyMount.setState({
            status: true
        }); 
        bodyMount.update();
        // console.log(bodyMount.html()); 
        expect(toJson(bodyMount)).toMatchSnapshot();
        
    });

    it('Test openMenu  ', () => {
		const fff = bodyMount.children()
		.children()
		.children()
        .children()
        .children()
        .children();
        bodyMount.setState({
            open: false
        });
        // console.log(bodyMount.state().status);
        const spy = jest.spyOn(fff.instance(), 'openMenu' as any);
		const ddd = (fff.instance() as any).openMenu();
        expect(spy).toHaveBeenCalledTimes(1);
        // console.log(bodyMount.state().open);
    });
    
});