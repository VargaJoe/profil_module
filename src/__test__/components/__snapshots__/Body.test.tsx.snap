// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Body /> mount rendering when data is added Match to snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "REPLACE",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "tsknke",
          "pathname": "/login",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <withRouter(Connect(BodyComponent))
        loginState="Unauthenticated"
        userName="Visitor"
      >
        <Route
          render={[Function]}
        >
          <Connect(BodyComponent)
            history={
              Object {
                "action": "REPLACE",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "tsknke",
                  "pathname": "/login",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "key": "tsknke",
                "pathname": "/login",
                "search": "",
                "state": undefined,
              }
            }
            loginState="Unauthenticated"
            match={
              Object {
                "isExact": false,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            userName="Visitor"
          >
            <BodyComponent
              dispatch={[Function]}
              history={
                Object {
                  "action": "REPLACE",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "tsknke",
                    "pathname": "/login",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "key": "tsknke",
                  "pathname": "/login",
                  "search": "",
                  "state": undefined,
                }
              }
              loginState="Unauthenticated"
              match={
                Object {
                  "isExact": false,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              userLoginState="Unauthenticated"
              userName="Visitor"
            >
              <div
                className="content_to_right"
              >
                <Header>
                  <header
                    className="sn_header"
                  >
                    <div
                      className="sn_wrapp"
                    >
                      <div
                        className="sn_header__search"
                      >
                        <input
                          type="text"
                        />
                        <button>
                          <i
                            className="fi flaticon-magnifier"
                          />
                        </button>
                      </div>
                    </div>
                  </header>
                </Header>
                <Sidebar
                  openMenu={[Function]}
                >
                  <div
                    className="sn_sidebar"
                  >
                    <LogoComponent
                      openMenu={[Function]}
                    >
                      <div
                        className="sn_sidebar__logo"
                      >
                        <Link
                          replace={false}
                          to="/"
                        >
                          <a
                            href="/"
                            onClick={[Function]}
                          >
                            <img
                              alt="site logo"
                              src="logo.png"
                            />
                          </a>
                        </Link>
                        <div
                          className="sn_menu"
                          onClick={[Function]}
                        >
                          <span />
                        </div>
                      </div>
                    </LogoComponent>
                    <Connect(SidebarUserComponent)>
                      <SidebarUserComponent
                        dispatch={[Function]}
                        user={Object {}}
                        userName="Visitor"
                      >
                        <Link
                          replace={false}
                          to="/user/Visitor"
                        >
                          <a
                            href="/user/Visitor"
                            onClick={[Function]}
                          >
                            <div
                              className="sn_sidebar__user"
                            >
                              <span
                                className="sn_sidebar__user__avatar--default"
                              >
                                <i
                                  className="fi flaticon-user"
                                />
                              </span>
                              <div
                                className="sn_sidebar__user__name"
                              />
                            </div>
                          </a>
                        </Link>
                      </SidebarUserComponent>
                    </Connect(SidebarUserComponent)>
                    <Connect(MenuComponent)>
                      <MenuComponent
                        getMenuItems={[Function]}
                        logout={[Function]}
                        userLoginState="Unauthenticated"
                      />
                    </Connect(MenuComponent)>
                  </div>
                </Sidebar>
                <div
                  className="sn_overflow"
                />
                <main
                  className="sn_main"
                >
                  <div
                    className="sn_wrapp"
                  >
                    <Switch />
                  </div>
                </main>
              </div>
            </BodyComponent>
          </Connect(BodyComponent)>
        </Route>
      </withRouter(Connect(BodyComponent))>
    </Provider>
  </Router>
</BrowserRouter>
`;

exports[`<Body /> mount rendering when data is added Match to snapshot 2`] = `
<div
  className="content_to_right"
>
  <Header />
  <Sidebar
    openMenu={[Function]}
  />
  <div
    className="sn_overflow"
  />
  <main
    className="sn_main"
  >
    <div
      className="sn_wrapp"
    >
      <Switch>
        <Route
          exact={true}
          path="/"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/otherUser"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/user/:user"
          render={[Function]}
          userName="Visitor"
        />
        <Route
          exact={true}
          path="/editUser"
          render={[Function]}
        />
        <Route
          key="usermanagement"
          path="/Usermanagement/:action/:guid"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/changePass"
          render={[Function]}
        />
        <Route
          key="usermanagement"
          path="/forgottenPassword"
          render={[Function]}
        />
      </Switch>
    </div>
  </main>
</div>
`;

exports[`<Body /> mount rendering when data is added Match to snapshot with status flase 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "tsknke",
          "pathname": "/login",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <withRouter(Connect(BodyComponent))
        loginState="Unauthenticated"
        userName="Visitor"
      >
        <Route
          render={[Function]}
        >
          <Connect(BodyComponent)
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "tsknke",
                  "pathname": "/login",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "key": "tsknke",
                "pathname": "/login",
                "search": "",
                "state": undefined,
              }
            }
            loginState="Unauthenticated"
            match={
              Object {
                "isExact": false,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            userName="Visitor"
          >
            <BodyComponent
              dispatch={[Function]}
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "tsknke",
                    "pathname": "/login",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "key": "tsknke",
                  "pathname": "/login",
                  "search": "",
                  "state": undefined,
                }
              }
              loginState="Unauthenticated"
              match={
                Object {
                  "isExact": false,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              userLoginState="Unauthenticated"
              userName="Visitor"
            >
              <div
                className="content_to_right"
              >
                <Header>
                  <header
                    className="sn_header"
                  >
                    <div
                      className="sn_wrapp"
                    >
                      <div
                        className="sn_header__search"
                      >
                        <input
                          type="text"
                        />
                        <button>
                          <i
                            className="fi flaticon-magnifier"
                          />
                        </button>
                      </div>
                    </div>
                  </header>
                </Header>
                <Sidebar
                  openMenu={[Function]}
                >
                  <div
                    className="sn_sidebar"
                  >
                    <LogoComponent
                      openMenu={[Function]}
                    >
                      <div
                        className="sn_sidebar__logo"
                      >
                        <Link
                          replace={false}
                          to="/"
                        >
                          <a
                            href="/"
                            onClick={[Function]}
                          >
                            <img
                              alt="site logo"
                              src="logo.png"
                            />
                          </a>
                        </Link>
                        <div
                          className="sn_menu"
                          onClick={[Function]}
                        >
                          <span />
                        </div>
                      </div>
                    </LogoComponent>
                    <Connect(SidebarUserComponent)>
                      <SidebarUserComponent
                        dispatch={[Function]}
                        user={Object {}}
                        userName="Visitor"
                      >
                        <Link
                          replace={false}
                          to="/user/Visitor"
                        >
                          <a
                            href="/user/Visitor"
                            onClick={[Function]}
                          >
                            <div
                              className="sn_sidebar__user"
                            >
                              <span
                                className="sn_sidebar__user__avatar--default"
                              >
                                <i
                                  className="fi flaticon-user"
                                />
                              </span>
                              <div
                                className="sn_sidebar__user__name"
                              />
                            </div>
                          </a>
                        </Link>
                      </SidebarUserComponent>
                    </Connect(SidebarUserComponent)>
                    <Connect(MenuComponent)>
                      <MenuComponent
                        getMenuItems={[Function]}
                        logout={[Function]}
                        userLoginState="Unauthenticated"
                      />
                    </Connect(MenuComponent)>
                  </div>
                </Sidebar>
                <div
                  className="sn_overflow"
                />
                <main
                  className="sn_main"
                >
                  <div
                    className="sn_wrapp"
                  >
                    <Switch />
                  </div>
                </main>
              </div>
            </BodyComponent>
          </Connect(BodyComponent)>
        </Route>
      </withRouter(Connect(BodyComponent))>
    </Provider>
  </Router>
</BrowserRouter>
`;

exports[`<Body /> mount rendering when data is added Match to snapshot with status true 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "tsknke",
          "pathname": "/login",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <withRouter(Connect(BodyComponent))
        loginState="Unauthenticated"
        userName="Visitor"
      >
        <Route
          render={[Function]}
        >
          <Connect(BodyComponent)
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "tsknke",
                  "pathname": "/login",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "key": "tsknke",
                "pathname": "/login",
                "search": "",
                "state": undefined,
              }
            }
            loginState="Unauthenticated"
            match={
              Object {
                "isExact": false,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            userName="Visitor"
          >
            <BodyComponent
              dispatch={[Function]}
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "tsknke",
                    "pathname": "/login",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "key": "tsknke",
                  "pathname": "/login",
                  "search": "",
                  "state": undefined,
                }
              }
              loginState="Unauthenticated"
              match={
                Object {
                  "isExact": false,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              userLoginState="Unauthenticated"
              userName="Visitor"
            >
              <div
                className="content_to_right"
              >
                <Header>
                  <header
                    className="sn_header"
                  >
                    <div
                      className="sn_wrapp"
                    >
                      <div
                        className="sn_header__search"
                      >
                        <input
                          type="text"
                        />
                        <button>
                          <i
                            className="fi flaticon-magnifier"
                          />
                        </button>
                      </div>
                    </div>
                  </header>
                </Header>
                <Sidebar
                  openMenu={[Function]}
                >
                  <div
                    className="sn_sidebar"
                  >
                    <LogoComponent
                      openMenu={[Function]}
                    >
                      <div
                        className="sn_sidebar__logo"
                      >
                        <Link
                          replace={false}
                          to="/"
                        >
                          <a
                            href="/"
                            onClick={[Function]}
                          >
                            <img
                              alt="site logo"
                              src="logo.png"
                            />
                          </a>
                        </Link>
                        <div
                          className="sn_menu"
                          onClick={[Function]}
                        >
                          <span />
                        </div>
                      </div>
                    </LogoComponent>
                    <Connect(SidebarUserComponent)>
                      <SidebarUserComponent
                        dispatch={[Function]}
                        user={Object {}}
                        userName="Visitor"
                      >
                        <Link
                          replace={false}
                          to="/user/Visitor"
                        >
                          <a
                            href="/user/Visitor"
                            onClick={[Function]}
                          >
                            <div
                              className="sn_sidebar__user"
                            >
                              <span
                                className="sn_sidebar__user__avatar--default"
                              >
                                <i
                                  className="fi flaticon-user"
                                />
                              </span>
                              <div
                                className="sn_sidebar__user__name"
                              />
                            </div>
                          </a>
                        </Link>
                      </SidebarUserComponent>
                    </Connect(SidebarUserComponent)>
                    <Connect(MenuComponent)>
                      <MenuComponent
                        getMenuItems={[Function]}
                        logout={[Function]}
                        userLoginState="Unauthenticated"
                      />
                    </Connect(MenuComponent)>
                  </div>
                </Sidebar>
                <div
                  className="sn_overflow"
                />
                <main
                  className="sn_main"
                >
                  <div
                    className="sn_wrapp"
                  >
                    <Switch />
                  </div>
                </main>
              </div>
            </BodyComponent>
          </Connect(BodyComponent)>
        </Route>
      </withRouter(Connect(BodyComponent))>
    </Provider>
  </Router>
</BrowserRouter>
`;
